winml/lib/Common/inc/onnx.h

```diff
@@ -3,7 +3,15 @@
 
 #pragma once
 
-#include "common.h"
+// STL
+#define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
+#define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING
+
+#ifdef OPTIONAL
+#undef OPTIONAL
+#endif
+
+#pragma warning(disable : 4100)
```

Comment:
Push, then pop warnings at end of header so it doesn't affect whoever included it (or push/pop around the code that triggers the warning)

Reply:
The think the point may be to disable the warning for all the files that include this header I believe. See the usage in common.h.

Reply:
Could you please fix the warning instead of ignore it? 

e.g. 
change
```c++
void foo(int x){}
```
to 
```c++
void foo(int){}
```

Reply:
When removing this warning ignore, there is an implementation of SetDmlOperator in the DML EP that returns E_NOTIMPL and thus generates this warning because the passed in parameters aren't used. Changing DML EP seems out of scope of this PR. To clarify, ignoring warning 4100 is not new to this PR, it is already in winrt_headers.h, the only difference is that winml_adapter now calls onnx.h instead of cpp_winrt.h so the warning suppression needed to be added to onnx.h as well. I'll remove the warning suppression in winrt_headers.h since it is not needed anymore to make this more explicit.